java/arcs/android/util/ProtoPrefetcher.kt

```diff
@@ -0,0 +1,141 @@
+        CrdtEntityProto.Operation::getDefaultInstance,
+        CrdtEntityProto.Operation.newBuilder()::build,
+        CrdtEntityProto.Operation.SetSingleton::getDefaultInstance,
+        CrdtEntityProto.Operation.SetSingleton.newBuilder()::build,
+        CrdtEntityProto.Operation.ClearSingleton::getDefaultInstance,
+        CrdtEntityProto.Operation.ClearSingleton.newBuilder()::build,
+        CrdtEntityProto.Operation.AddToSet::getDefaultInstance,
+        CrdtEntityProto.Operation.AddToSet.newBuilder()::build,
+        CrdtEntityProto.Operation.RemoveFromSet::getDefaultInstance,
+        CrdtEntityProto.Operation.RemoveFromSet.newBuilder()::build,
+        CrdtEntityProto.Operation.ClearAll::getDefaultInstance,
+        CrdtEntityProto.Operation.ClearAll.newBuilder()::build,
+        CrdtSetProto::getDefaultInstance,
+        CrdtSetProto.newBuilder()::build,
+        CrdtSetProto.DataValue::getDefaultInstance,
+        CrdtSetProto.DataValue.newBuilder()::build,
+        CrdtSetProto.Data::getDefaultInstance,
+        CrdtSetProto.Data.newBuilder()::build,
+        CrdtSetProto.Operation::getDefaultInstance,
+        CrdtSetProto.Operation.newBuilder()::build,
+        CrdtSetProto.Operation.Add::getDefaultInstance,
+        CrdtSetProto.Operation.Add.newBuilder()::build,
+        CrdtSetProto.Operation.Remove::getDefaultInstance,
+        CrdtSetProto.Operation.Remove.newBuilder()::build,
+        CrdtSetProto.Operation.FastForward::getDefaultInstance,
+        CrdtSetProto.Operation.FastForward.newBuilder()::build,
+        CrdtSingletonProto::getDefaultInstance,
+        CrdtSingletonProto.newBuilder()::build,
+        CrdtSingletonProto.Data::getDefaultInstance,
+        CrdtSingletonProto.Data.newBuilder()::build,
+        CrdtSingletonProto.Operation::getDefaultInstance,
+        CrdtSingletonProto.Operation.newBuilder()::build,
+        CrdtSingletonProto.Operation.Update::getDefaultInstance,
+        CrdtSingletonProto.Operation.Update.newBuilder()::build,
+        CrdtSingletonProto.Operation.Clear::getDefaultInstance,
+        CrdtSingletonProto.Operation.Clear.newBuilder()::build,
+        ReferencableProto::getDefaultInstance,
+        ReferencableProto.newBuilder()::build,
+        ReferencableSetProto::getDefaultInstance,
+        ReferencableSetProto.newBuilder()::build,
+        RawEntityProto::getDefaultInstance,
+        RawEntityProto.newBuilder()::build,
+        CrdtEntityReferenceProto::getDefaultInstance,
+        CrdtEntityReferenceProto.newBuilder()::build,
+        ReferenceProto::getDefaultInstance,
+        ReferenceProto.newBuilder()::build,
+        ReferencablePrimitiveProto::getDefaultInstance,
+        ReferencablePrimitiveProto.newBuilder()::build,
+        VersionMapProto::getDefaultInstance,
+        VersionMapProto.newBuilder()::build
+    )
+
+    /**
+     * Execute [procedures] to warm up Arcs protos ahead-of-time.
+     */
+    operator fun invoke() = prefecher.execute {
```

Comment:
I'd suggest replacing this with just a `ProtoPrefetcher.prefetch()` method. It's kind of strange invoking `ProtoPrefetcher()` as a function.

Reply:
done